compiler/rustc_mir_transform/src/patch.rs

```diff
@@ -192,6 +192,12 @@ impl<'tcx> MirPatch<'tcx> {
         Local::new(index)
     }
 
+    /// Remove the last temporary local in the queue.
+    /// CAN ONLY DO THIS when the local is unused.
+    pub(crate) fn revert_last_new_temp(&mut self) {
+        self.new_locals.pop();
+    }
+
```

Comment:
I'd be more confortable if we did not have this method. It does not check that we are popping the local that the caller expects (what it there is some new temp in between?), and I'm not convinced having an extra local is an issue. It will be cleaned up by `SimplifyLocals` anyway won't it?